src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Helpers/PolicyHelper.cs

```diff
@@ -41,5 +46,164 @@ public static List<string> GetRolesWithAccess(XacmlPolicy policy)
+                    {
+                        resourcePolicy.Actions.Add(resourceAction);
+                    }
+                    else
+                    {
+                        ResourceAction existingAction = resourcePolicy.Actions.First(action => action.Match.Value == resourceAction.Match.Value && action.Match.Id == resourceAction.Match.Id);
+                        existingAction.RoleGrants.AddRange(resourceAction.RoleGrants.Where(roleGrant => !existingAction.RoleGrants.Any(existingRoleGrant => existingRoleGrant.RoleTypeCode == roleGrant.RoleTypeCode)));
+                    }
+                }
+            }
+        }
+
+        private static List<ResourceAction> GetActionsFromRule(XacmlRule rule, List<RoleGrant> roles)
+        {
+            List<ResourceAction> actions = new List<ResourceAction>();
+            foreach (XacmlAnyOf anyOf in rule.Target.AnyOf)
+            {
+                foreach (XacmlAllOf allOf in anyOf.AllOf)
+                {
+                    AttributeMatch actionAttributeMatch = new AttributeMatch();
+                    foreach (XacmlMatch xacmlMatch in allOf.Matches)
+                    {
+                        if (xacmlMatch.AttributeDesignator.Category.Equals(XacmlConstants.MatchAttributeCategory.Action))
+                        {
+                            actionAttributeMatch.Id = xacmlMatch.AttributeDesignator.AttributeId.OriginalString;
+                            actionAttributeMatch.Value = xacmlMatch.AttributeValue.Value;
+                            ResourceAction resourceAction = new ResourceAction
+                            {
+                                Match = actionAttributeMatch,
+                                RoleGrants = new List<RoleGrant>(),
+                                Title = new LocalizedText(xacmlMatch.AttributeValue.Value, xacmlMatch.AttributeValue.Value, xacmlMatch.AttributeValue.Value)
+                            };
+                            resourceAction.RoleGrants.AddRange(roles);
+                            if (!actions.Contains(resourceAction))
+                            {
+                                actions.Add(resourceAction);
+                            }
+                        }
+                    }
+                }
+            }
+
+            return actions;
+        }
+
+        private static List<RoleGrant> GetRolesFromRule(XacmlRule rule)
+        {
+            List<RoleGrant> roles = new List<RoleGrant>();
+            foreach (XacmlAnyOf anyOf in rule.Target.AnyOf)
+            {
+                foreach (XacmlAllOf allOf in anyOf.AllOf)
+                {
+                    foreach (XacmlMatch xacmlMatch in allOf.Matches)
+                    {
+                        if (xacmlMatch.AttributeDesignator.Category.Equals(XacmlConstants.MatchAttributeCategory.Subject))
```

Comment:
```suggestion
                        if (xacmlMatch.AttributeDesignator.Category.Equals(XacmlConstants.MatchAttributeCategory.Subject) && xacmlMatch.AttributeDesignator.AttributeId.Equals(XacmlRequestAttribute.RoleAttribute))
```
This should filter out none-roletype subjects as @TheTechArch pointed out.